integration-tests/src/tests/client/flat_storage.rs

```diff
@@ -494,3 +495,58 @@ fn test_flat_storage_iter() {
         }
     }
 }
+
+#[test]
+fn test_not_supported_block() {
+    init_test_logger();
+    let genesis = Genesis::test(vec!["test0".parse().unwrap()], 1);
+    let shard_layout = ShardLayout::v0_single_shard();
+    let shard_uid = shard_layout.get_shard_uids()[0];
+    let store = create_test_store();
+
+    let mut env = setup_env(&genesis, store.clone());
+    // Produce blocks up to `START_HEIGHT`.
+    for height in 1..START_HEIGHT {
+        env.produce_block(0, height);
+    }
+
+    // Trie key which must exist in the storage.
+    let trie_key_bytes =
+        near_primitives::trie_key::TrieKey::Account { account_id: "test0".parse().unwrap() }
+            .to_vec();
+    // Create trie, which includes creating chunk view, and get `ValueRef`s
+    // for post state roots for blocks `START_HEIGHT - 3` and `START_HEIGHT - 2`.
+    // After creating the first trie, produce block `START_HEIGHT` which moves flat storage
+    // head 1 block further and invalidates it.
+    let get_ref_results: Vec<_> = (START_HEIGHT - 3..START_HEIGHT - 1)
+        .map(|height| {
+            let block_hash = env.clients[0].chain.get_block_hash_by_height(height).unwrap();
+            let state_root = *env.clients[0]
+                .chain
+                .get_chunk_extra(&block_hash, &ShardUId::from_shard_id_and_layout(0, &shard_layout))
+                .unwrap()
+                .state_root();
+
+            let trie = env.clients[0]
+                .runtime_adapter
+                .get_trie_for_shard(shard_uid.shard_id(), &block_hash, state_root, true)
+                .unwrap();
+            if height == START_HEIGHT - 3 {
+                env.produce_block(0, START_HEIGHT);
```

Comment:
nit: it is generally nice to avoid side effects in `.map`, so I would rewrite this as a loop
doesn't matter that much in tests, so feel free to keep it as it is